ml-conversational-analytic-tool/run.py

```diff
@@ -2,14 +2,30 @@
 # SPDX-License-Identifier: Apache-2.0
 
 import argparse
+import tarfile
+import os
+
 from sklearn.model_selection import train_test_split
 
 from baseCNN import BaseCNN
 from baseLSTM import BaseLSTM
 from preProcessedDataset import PreProcessedDataset
 
-
-def run(annotated_filename, dataset_filename, outcome, encoding_type, model_type, padding):
+model_direcrtory = 'models'
+model_version = '0001'
+
+def save_model(model, name):
+    if not os.path.exists(model_direcrtory):
+        os.makedirs(model_direcrtory)
+    os.chdir(model_direcrtory)
+    model.saveModel(name=name, version=model_version)
+    os.chdir(name)
```

Comment:
I think `.saveModel` can be called with passing the path? I don't think changing directories are necessary.

Reply:
The only reason to change the directory is to ensure the proper layout is saved in the archive. It should be 
```
<model_version>\
                assets\
                variables\
                <proto buffer files> 
```
May be there is some clever way to force tar to create that layout without changing the directories, but I do not know it :)

Reply:
I'm not sure if this is the best way, but you could also consider something like this.
I personally find it a bit more cleaner this way, but not a blocker for getting this merged :) 

```
model_path = "{}/{}".format(model_direcrtory, name)
tar_file_name = "{}-{}".format(name, model_version)
tar_path = "{}/{}.tar.gz".format(model_path, tar_file_name)

model.saveModel(name=model_path, version=model_version)
tar = tarfile.open(tar_path, "w:gz")
tar.add(model_path)
tar.close()
```

Reply:
I changed the strings to use `format`,  but cannot avoid using `chdir`. When running `tar` from the top directory, the structure of the archive is:
```
 models\
     <name>\
         <version>\
                 assets\
                 variables\
                 <proto buffer files> 
```
which is not expected from the  prediction  service and it fails to load the model.